dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md

```diff
@@ -0,0 +1,105 @@
+
+     :::code language="csharp" source="snippets/work-with-data-grid/csharp/form1.cs" id="find_the_record":::
+     :::code language="vb" source="snippets/work-with-data-grid/vb/form1.vb" id="find_the_record":::
+
+## How to ensure the selected row in a child table remains at the correct position
+
+When you work with data binding in Windows Forms, you'll display data in a parent/child or master/details view. It's a data-binding scenario where data from the same source is displayed in two controls. Changing the selection in one control causes the data displayed in the second control to change. For example, the first control might contain a list of customers and the second a list of orders related to the selected customer in the first control.
+
+When you display data in a parent/child view, you might have to take extra steps to ensure that the currently selected row in the child table isn't reset to the first row of the table. In order to do this, you'll have to cache the child table position and reset it after the parent table changes. Typically the child reset occurs the first time a field in a row of the parent table changes.
+
+### To cache the current child position
+
+01. Declare an integer variable to store the child list position and a Boolean variable to store whether to cache the child position.
+
+     :::code language="csharp" source="snippets/work-with-data-grid/csharp/form2.cs" id="store_the_child_list_pos":::
+     :::code language="vb" source="snippets/work-with-data-grid/vb/form2.vb" id="store_the_child_list_pos":::
+
+01. Handle the <xref:System.Windows.Forms.CurrencyManager.ListChanged> event for the binding's <xref:System.Windows.Forms.CurrencyManager> and check for a <xref:System.ComponentModel.ListChangedType> of <xref:System.ComponentModel.ListChangedType.Reset>.
+
+01. Check the current position of the <xref:System.Windows.Forms.CurrencyManager>. If it's greater than first entry in the list (typically 0), save it to a variable.
+
+     :::code language="csharp" source="snippets/work-with-data-grid/csharp/form2.cs" id="check_current_pos_of_currency_manager":::
+     :::code language="vb" source="snippets/work-with-data-grid/vb/form2.vb" id="check_current_pos_of_currency_manager":::
+
+01. Handle the parent list's <xref:System.Windows.Forms.BindingManagerBase.CurrentChanged> event for the parent currency manager. In the handler, set the Boolean value to indicate it isn't a caching scenario. If the `CurrentChanged` occurs, the change to the parent is a list position change and not an item value change.
+
+     :::code language="csharp" source="snippets/work-with-data-grid/csharp/form2.cs" id="handle_current_change_event":::
+     :::code language="vb" source="snippets/work-with-data-grid/vb/form2.vb" id="handle_current_change_event":::
+
+### To reset the child position
+
+01. Handle the <xref:System.Windows.Forms.BindingManagerBase.PositionChanged> event for the child binding's <xref:System.Windows.Forms.CurrencyManager>.
```

Comment:
Handle the <xref:System.Windows.Forms.BindingManagerBase.PositionChanged> event for the child **table** binding's....